src/rtpi_internal.h

```diff
@@ -6,22 +6,43 @@
 
 #include <linux/futex.h>
 
-#include "rtpi.h"
-
-typedef struct pi_mutex {
-	__u32	futex;
-	__u32	flags;
-} pi_mutex_t;
-
-typedef struct pi_cond {
-	__u32		cond;
-	__u32		flags;
-
-	pi_mutex_t	priv_mut;
-	__u32		wake_id;
-	__u32		pending_wake;
-	__u32		pending_wait;
-	pi_mutex_t	*mutex;
-} pi_cond_t;
+/*
+ * PI Mutex
+ */
+union pi_mutex {
+	struct {
+		__u32	futex;
+		__u32	flags;
+	};
+	__u32 pad[4];
```

Comment:
What led to this specific pad size? Let's be sure to explicitly call out the rationale.

Reply:
I was unsure how much padding to add so I've used double the current structure size as a metric.

An alternative would be to make it cache line sized (and aligned). The difficulty with this approach would be figuring out at build time the cache line size (or we could default it to the x86_64, 64 bytes size). Thoughts?

Reply:
I will post a new version of this pull request which makes the padding 64 bytes and sets the alignment to 64 i.e. one cache line on most 64 bit machines. This should give us better performance/reasoning for the padding size.

Not sure what the best work flow is for updating this github PR so I will post it as a new pull request and we can close this one if we're happy with the new version. I can also force push and update to the current PR branch if that's preferable.